Tests/smile_counter_dev.py

```diff
@@ -49,32 +58,49 @@ def main():
         face = face_cascade.detectMultiScale(gray, scaleFactor=1.1, minNeighbors=4)
         # Draw the rectangle around each face
         for (x, y, w, h) in face:
-            img = cv2.rectangle(img, (x,y), (x+w, y+h), (0, 0, 255), 3)
+            img = cv2.rectangle(img, (x, y), (x + w, y + h), (0, 0, 255), 3)
+            # Crop to detected face
+            cropped_face_gray = gray[y:y+h, x:x+w]
             # Detect the Smile
-            smile = smile_cascade.detectMultiScale(gray, scaleFactor=1.4, minNeighbors=10)
-            if len(smile) > 0:
-                logger.info("Smile detected in photo with filename: " + photo_name)
-                detected_smiles += 1
-            else:
-                logger.info("Do not detected smile in photo with filename: " + photo_name)
+            smile = smile_cascade.detectMultiScale(cropped_face_gray, scaleFactor=1.8, minNeighbors=20)
+
             # Draw the rectangle around detected smile
-            for x, y, w, h in smile:
-                img = cv2.rectangle(img, (x, y), (x+w, y+h), (255, 0, 0), 3)
+            for x_, y_, w_, h_ in smile:
+                img = cv2.rectangle(img, (x_ + x, y_ + y), (x_ + w_  + x, y_ + h_ + y), (255, 0, 0), 3)
+                # Save cropped gray face
+                cropped_face_gray = cv2.rectangle(cropped_face_gray, (x_, y_), (x_ + w_, y_ + h_), (255, 0, 0), 3)
+                cropped_face_filename = "croppedface" + str(x_ + y_)
+                save_output_images(cropped_face_filename, cropped_face_gray)
+
+                if len(smile) > 0:
+                    logger.info("Smile detected in photo with filename: " + photo_name)
+                    detected_smiles += 1
+                else:
+                    logger.info("Do not detected smile in photo with filename: " + photo_name)
 
         # Display output
-        cv2.imshow('smile detect', img)
-        filename = _create_output_path(DEFAULT_OUTPUT_FOLDER_PATH, photo_name, DEFAULT_OUTPUT_FILE_EXTENSION)
-        cv2.imwrite(filename, img)
+        show_output_images(window_name = config_data['DEFAULT_IMAGE_WINDOW_NAME'], img = img)
 
-        # Stop if escape key is pressed
-        key = cv2.waitKey(30) & 0xff
-        if key == 27:
-            break
+        save_output_images(photo_name=photo_name, img=img)
 
-    # NOTE: Logger should work like print if appropriate log level is set.
     logger.info("Analyzed " + str(len(photos)) + " photos")
     logger.info("Detected " + str(detected_smiles) + " smiles")
+
+
+def time_counting(func):
```

Comment:
Functions should probably be named with a verb rather than using the gerundium form. 
( https://llego.dev/posts/writing-clean-pep-8-compliant-code-better-collaboration/#use-verbs-for-function-names )

So, for example, count_time() instead of time_counting()

According to clean code rules, this also doesn't say much, so it would be better to call it, for example, count_total_function_execution_time( function ).

"as the scope (of function) decreases, the names start to get longer"

https://bpoplauschi.github.io/2021/01/30/Clean-Code-Naming-by-Uncle-Bob-part-3.html

PS: it would probably be better to simply run the function to measure time before calling the function to be measured and after calling it, end the time measurement, instead of passing a reference to the function as a parameter and running it inside another function (here, to measure time).